### DISCUSSION TITLE
Color logs based on span data

### DISCUSSION THREAD

**Discussion by wdv4758h** [tokio-rs/tracing#1639] [Q&A] [answered]
Assuming I create span with variable `id`. And I want different color for different `id` in logs. For example, `id=1` is green and `id=2` is yellow. So it is easier to identify related logs visually by color.

Is this possible without too much hack?

**Comment by davidbarsky** [comment, marked as answer]
You might need to create your own types implementing [`FormatEvent`](https://docs.rs/tracing-subscriber/0.2.25/tracing_subscriber/fmt/trait.FormatEvent.html) and [`FormatFields`](https://docs.rs/tracing-subscriber/0.2.25/tracing_subscriber/fmt/trait.FormatFields.html) to color your logs depending on the data.

**Reply by wdv4758h** [reply]
Thanks for the pointer!
